step-templates/octopus-validate-deploying-user.json

```diff
@@ -0,0 +1,54 @@
+{
+    "Id": "22c55424-bdcb-44aa-a3a1-978e607f6e2f",
+    "Name": "octopus-validate-deploying-user",
+    "Description": "Verifies current deploying user didn't deploy previously to specified environment.",
+    "ActionType": "Octopus.Script",
+    "Version": 1,
+    "CommunityActionTemplateId": null,
+    "Packages": [],
+    "Properties": {
+      "Octopus.Action.RunOnServer": "false",
+      "Octopus.Action.Script.ScriptSource": "Inline",
+      "Octopus.Action.Script.Syntax": "PowerShell",
+      "Octopus.Action.Script.ScriptBody": "$releaseId = $OctopusParameters[\"Octopus.Release.Id\"]\n$currentDeployerId = $OctopusParameters[\"Octopus.Deployment.CreatedBy.Id\"]\n$userName = $OctopusParameters[\"Octopus.Deployment.CreatedBy.Username\"]\n$environmentId = $OctopusParameters[\"Octopus.Environment.Id\"]\n$spaceId = $OctopusParameters[\"Octopus.Space.Id\"]\n\n$header = @{ \"X-Octopus-ApiKey\" = $octopusAPIKey }\n\n$deploymentDetails = (Invoke-RestMethod -Method Get -Uri \"$octopusURL/api/$($spaceId)/releases/$($releaseId)/deployments/\" -Headers $header)\n\n# Get details for deployment to preceding environment\n$allEnvironments = (Invoke-RestMethod -Method Get -Uri \"$octopusURL/api/$($spaceId)/environments\" -Headers $header)\n$environmentItem = $allEnvironments.Items | where-object { $_.Name -eq $precedingEnvironment }\n$environmentId = $environmentItem.Id\n\n# Load all deploys to the previous environment\n$environmentDeploys = $deploymentDetails.Items | Where-Object {$_.EnvironmentId -eq $environmentId}\n\n# Iterate deployments to the previous environment to validate current deployer\nforeach($prevdeployment in $environmentDeploys)\n    {\n    \tif($prevDeployment.Id -eq $OctopusParameters[\"Octopus.Deployment.Id\"])\n        {continue}\n    \telse\n        {\n    \t\tif($prevdeployment.DeployedById -eq $currentDeployerId )\n        \t{\n            \tWrite-Highlight \"$userName previously deployed this project to $precedingEnvironment - deployment cancelled.\"\n            \tThrow \"$userName previously deployed this project to $precedingEnvironment - deployment cancelled.\"\n        \t}\n        }\n    }"
```

Comment:
In your script I see you referencing the _$octopusURL_, _$octopusAPIKey_, _$precedingEnvironment_ variables but they aren't being mapped to your parameters anywhere. You'd need to do the same thing that you're doing with those system variables at the top of the script to make those work.

`$octopusURL = $OctopusParameters["ovdu_OctopusURL"]`
`$octopusAPIKey = $OctopusParameters["ovdu_OctopusAPIKey"]`
`$precedingEnvironment = $OctopusParameters["ovdu_precedingEnvironment"]`